New Feature: Attribute driven users

DavidSie opened the pull request:
### Description

[Describe what this change achieves]
### Issues Resolved

[List any existing issues this PR resolves]
### Check List
- [x] All tests pass. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] The CLA has been signed. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD

This commit gives an ability to choose some of the users from the group that you want to install.
I order to do so one must set node['users']['list'] for example to ['sysadmin_2'].
This will cause that from all other users only sysadmin_2 will be installed( if exist as databag)

chef-supermarket commented:
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

jblaine commented on README.md:
```diff
@@ -7,6 +7,23 @@ Manages OS users from databags.
+  <tr>
+    <td><tt>['users']['list']</tt></td>
+    <td>Array</td>
+    <td> If not empty only users from list will be created </td>
```
"created" should be "acted on"

jblaine commented on README.md:
```diff
@@ -63,6 +80,8 @@ end
 
 **Note**: If you do not specify the data_bag, the default will be to look for a databag called users.
 
+**Note**: If you want to install only part of your users that belong to the group you can specify them in the attribute  `node['users']['list']`
```
"install" should be "act on"

DavidSie commented:
I changed in readme from `create` to `act on`  but I implemented this only for create action since I believe remove action works well with a current flow

iennae commented:
Hey @DavidSie,

Thanks for this contribution. A few notes about this

1.  Please don't use html in the readme, as this breaks the parsing on Supermarket when released.
2. Please don't bump the version in metadata.rb. We'll do that when we release the cookbook.
3. Can you make it more explicit about use cases expected for this? The description in the README doesn't provide enough context.

Also something seems to be causing extra changes to appear that don't look like they are real changes. Take a look at the Files Changed tab for providers/manage.rb. Any idea what's going on there?

tas50 commented:
Sorry for the late response, but I'm going to close this PR out. The goal of this cookbook is to provide an opinionated method for setting up users from data bags. Chef itself already provides the user resource which can be used with an attribute to create users. This really seems out of scope and adds complexity to the cookbook interface.  See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/COOKBOOK_TEAM.MD#support-the-95-use-case
